### ISSUE TITLE
struct-tag rule false-positive on json inline tag

### ISSUE THREAD

**Issue by finnan444** [mgechev/revive#520]
**Describe the bug**
[struct-tag](https://github.com/mgechev/revive/blob/master/RULES_DESCRIPTIONS.md#struct-tag) produces warning for json inline tag `struct-tag: unknown option 'inline' in JSON tag (revive)`

**To Reproduce**
1. add 
```golang
type Beta struct {
    Alpha `json:",inline"`
}
```
2. Run revive with struct-tag rule enabled

**Comment by chavacava** [comment]
Hi @finnan444 thanks for filling the issue.
In fact `inline` is not an [official option of JSON struct tags](https://pkg.go.dev/encoding/json?utm_source=godoc) (this [8 years old but still ongoing discussion](https://github.com/golang/go/issues/6213) also talks about it)

(Actually, the `inline` option works by lucky it might be possible that if you change yours `inline` by `something` it will still working)

**Comment by finnan444** [comment]
Hi @chavacava, sorry for inconvenience, this tag is widely used in my company (this practice came from kubernetes repo), so i thought `inline` is part of standard now

**Comment by chavacava** [comment]
Do not worry, we have been all trapped by this `inline` option :)

Thanks again for taking the time of filling the issue.

**Comment by s3rj1k** [comment]
@chavacava So why not add a configuration option to struct-tag linter?

**Comment by chavacava** [comment]
@s3rj1k What do you need to configure?

**Comment by s3rj1k** [comment]
I was thinking of an array with additionally allowed list of tags, to make that `inline` tag allowed per user-configuration

**Comment by chavacava** [comment]
Could you please develop on the interest of using non-standard options in struct tags?
Why not just remove `inline` from JSON tags? (unknown options are ignored by `encoding/json`)

**Comment by s3rj1k** [comment]
@chavacava inline tag is needed for kubernetis CRD creation, please see link below
https://book.kubebuilder.io/reference/generating-crd.html#additional-printer-columns

**Comment by chavacava** [comment]
I do not know about CRD creation but I know that non standard tag options are [ignored by `encoding\json`](https://github.com/golang/go/blob/d2552037426fe5a190c74172562d897d921fe311/src/encoding/json/encode.go#L1213) so its hard to me to imagine what is the interest of using `inline` option. 
Did you try to replace `json:",inline"` by just `json:""` when creating a CRD? It should be semantically equivalent as [this code](https://go.dev/play/p/mV_tKsxFa6c) shows

> I was thinking of an array with additionally allowed list of tags, to make that inline tag allowed per user-configuration

If we add a configuration, it should allow additional options but scoped to particular tags. For example, by allowing  `inline` for JSON tags does not allow it for other tags (ASN, csv, base64, ...)

**Comment by s3rj1k** [comment]
@chavacava inline is used across all K8s code, for example here https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/garbagecollector/metaonly/types.go#L30

I would rather use it also like docs say.

> If we add a configuration, it should allow additional options but scoped to particular tags. For example, by allowing inline for JSON tags does not allow it for other tags (ASN, csv, base64, ...)

Sure, so a list per tag or a map[string][]string

**Comment by chavacava** [comment]
>  inline is used across all K8s code

I know. I use Kubernetes repo as the main sandbox for testing new linting rules because: a) it is a huge repo and, b) it is not famous for its coding style quality ;)

**Comment by s3rj1k** [comment]
So, either we have some kind of a configurable option for struct-tag, to allow this K8s mess and have other struct tags checked or we go and disable struct-tag linter for specific places or for all code base.

Personally I would be happy with configuration option, that way linter will work on other structs inside project.

**Comment by s3rj1k** [comment]
@chavacava Bump, I am forced to use exclude rule for this and this is ugly.

**Comment by chavacava** [comment]
@s3rj1k I've made the PR #802 to let users tag structs with all the unofficial options they can imagine.
